### PR TITLE
Ensure thread safety of fiss sessions

### PR DESCRIPTION
**Pull request by xbrianh**
Depends on #245 #244 #243

### CODE REVIEW

**Review by DailyDreaming — APPROVED** [review]
LGTM.

File: terra_notebook_utils/table.py [hunk 1]
```diff
@@ -1,10 +1,186 @@
...
+Row = namedtuple("Row", "name attributes")
+
+VALUE = Union[str, int, float, bool]
+ATTRIBUTES = Dict[str, Union[VALUE, Iterable[VALUE]]]
+UPDATE_OPS = List[Dict[str, Any]]
+COLUMN_HEADERS = Tuple[str, ...]
+ROW_LIKE = Tuple[str, ATTRIBUTES]
+
+# A note on types:
+# As presented in Terra's UI, tables may contain string, integer, and booleas values.  However, Firecloud API clients
```
Comment (DailyDreaming) [inline terra_notebook_utils/table.py:25]: `booleas`

File: terra_notebook_utils/table.py [hunk 2]
```diff
@@ -1,10 +1,186 @@
...
+# must transmit tables using TSV formatted data, which is not typed. So, alas, we cannot transmit typed data to Terra
+# data tables, and cannot transform uploaded types via the Firecloud API (as far as I know).
+#
+# It would be preferable if the Firecloud API exposed JSON endpoints for uploading table data (which may already exist
+# on the backedn?)
```
Comment (DailyDreaming) [inline terra_notebook_utils/table.py:30]: `backedn`

**Review by DailyDreaming — APPROVED** [review]
LGTM.  Two nits.

File: terra_notebook_utils/table.py (hunk 1 above)
Comment (xbrianh) [reply terra_notebook_utils/table.py:25]: will fix

File: terra_notebook_utils/table.py (hunk 2 above)
Comment (xbrianh) [reply terra_notebook_utils/table.py:30]: will fix

**Pull request merged**